libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2167,8 +2167,26 @@ string YulUtilFunctions::readFromStorage(Type const& _type, size_t _offset, bool
 
 string YulUtilFunctions::readFromStorageDynamic(Type const& _type, bool _splitFunctionTypes)
```

Comment:
I'm a bit worried that this might lead to unnecessary copies from storage to memory. Can you check that it is always used properly?

Reply:
I saw some code in the form of
```
			structMembers[i].type->isValueType() ?
				readFromStorageDynamic(*structMembers[i].type, true) :
				readFromStorage(*structMembers[i].type, memberStorageOffset, true)
```
Can this now be simplified?

Reply:
It can be simplified, yes. Will do it

Reply:
Can be simpler, but point was that in dynamic we have additional check (if offset is 0 it will call panic()). So, In simpler version it look nicer, but it will have this check which probably optimizer can remove.